droid-command-line/src/main/java/uk/gov/nationalarchives/droid/command/action/ProfileRunCommand.java

```diff
@@ -107,19 +114,52 @@ public void onProgress(Integer progress) {
         
     }
 
+    /**
+     * @return The default binary and container signatures, but will override with different ones if provided.
+     * @throws SignatureFileException if there's a problem obtaining the signature files.
+     */
+    private Map<SignatureType, SignatureFileInfo> getSignatureFiles() throws SignatureFileException {
+        Map<SignatureType, SignatureFileInfo> sigs = signatureManager.getDefaultSignatures();
+        if (binarySignaturesFileName != null) {
```

Comment:
binarySignatureFile could be empty, if it is set as empty from outside of this class, then `Paths.get` might throw an exception

Reply:
True.  Or if it simply has an incorrect file name.  `Paths.get()` would throw an `InvalidPathException`, which is a subtype of RuntimeException, and propagate up to the `DroidCommandLine.processExecution()` method, which would catch it and output an error to the console.

Of course, the error may not be particularly helpful to the user.  A custom exception if we can't get a Path could provide better user feedback.

Reply:
The point of the null test isn't actually to avoid Path errors, btw.  If a specific signature file is not set, then we use the profile defaults which are already obtained.  If *something* is set (not null), then we try to use that instead.     I'm working on a tiny change that gives better user feedback if a supplied filename is invalid.

Reply:
That's cool, thanks 👍🏼

Reply:
Turns out it is impossible from the command line to set a blank signature file.  And we get an error if you try to use a non-blank but invalid signature file.  So I would argue we might not need to do anything here.